text/0507-subnets.md

```diff
@@ -0,0 +1,402 @@
+# Full control over VPC and subnet configuration
+
+* **Original Author(s):**: @otaviomacedo
+* **Tracking Issue**: #507
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+New L2 constructs to allow for greater control over VPC design. Among other
+things, these constructs allow users to create and configure subnets in
+ways that are not currently possible with the `Vpc` construct.
+
+## Working Backwards
+
+### IP addressing
+
+When you create a VPC, in addition to the primary IP block, you can provide
+one or more secondary IP blocks, by providing them in the CIDR format:
+
+```ts
+const vpc = new Vpc(this, 'vpc', {
+  // Primary address block
+  ipAddresses: IpAddresses.ipv4('10.1.0.0/16'),
+});
+
+// The secondary address block must be in the same RFC 1918 range
+vpc.addSecondaryAddressBlock(IpAddresses.ipv4('10.2.0.0/16'));
```

Comment:
I imagine there will also be a `secondaryAddressBlocks?: IpAddresses[]` property?

Reply:
Yes, we can add that, too.